dags/databricks-model-serve-sagemaker-example.py

```diff
@@ -0,0 +1,146 @@
+from airflow.decorators import task, dag
+from airflow.operators.python import ShortCircuitOperator
+from mlflow.sagemaker import SageMakerDeploymentClient
+from mlflow.tracking import MlflowClient
+from pendulum import datetime
+
+from include.sample_data import test_sample
+
+docs = """
+Demonstrates orchestrating ML model serving pipelines executed on Databricks with Airflow.
+
+Checks if there is a model marked for `Staging` that does not have a `deployed: True` tag in the registry. If that
+condition is met, it proceeds to deploy the new model version to Sagemaker and create an end point. Once the end point
+is created a test on the endpoint with sample data is performed and if it gets a successful response then the model
+in the registry is tagged as `deployed: True`.
+"""
+
+model_name = 'census_pred'
+region = 'us-east-2'
+model_uri = f'models:/{model_name}/Staging'
+deployment_name = 'census-pred-deployment'
+target_uri = f'sagemaker:/{region}'
+
+
+@dag(
+    start_date=datetime(2022, 1, 1),
+    schedule_interval=None,
+    catchup=False,
+    doc_md=docs
+)
+def databricks_model_serve_sagemaker_example():
+
+    @task(multiple_outputs=True)
+    def check_model_version_status():
+        """Check if a new model version is available
+
+        Collect info if there is a new model version that has been transitioned to Staging and determine its deployment
+        status.
+
+        Returns JSON with deployed status, version number, and model description.
+        """
+        client = MlflowClient()
+
+        # Iterate through all model versions and get only the one market for Staging.
+        # Then gather info on its version number, deployment tag info, and description.
+        for mv in client.search_model_versions(f"name='{model_name}'"):
+            model_version = dict(mv)
+            if model_version['current_stage'] == 'Staging':
+                logging.info(pformat(mv, indent=4))
+
+                version_number = model_version['version']
+                description = model_version['description']
+
+                if 'deployed' not in model_version['tags']:
+                    deployed = False
+                elif 'deployed' in model_version['tags'] and model_version['tags']['deployed'] == 'False':
+                    deployed = False
+                else:
+                    deployed = True
+
+                return {'deployed': deployed, 'version': version_number, 'description': description}
+
+        # If no model is marked for Staging, then return "Not Available".
+        return {'deployed': 'Not Available', 'version': None, 'description': None}
+
+    model_status_result = check_model_version_status()
+
+    # Based on the output of the check_model_version_status() task determine if a new model needs to be deployed then
+    # proceed accordingly.
+    new_version_confirmation = ShortCircuitOperator(
+        task_id="new_version_confirmation",
+        python_callable=lambda deployed: not deployed,
```

Comment:
Should this also handle the case when `deployed` == "Not Available"?

Reply:
`not deployed` for the value "Not Available" gets evaluated as `False` which is expected behavior. I could have just returned a boolean instead for the situation where "Not Available" is returned, but I wanted it to be more explicit for logging and traceability of what is happening.